### ISSUE TITLE
Cannot extract image with starting point 0,0

### ISSUE THREAD

**Issue by tostrowski** [h2non/bimg#106]
I think it's a valid use case to extract image with no offset, that is with a starting point Top: 0, Left: 0. Current design does not perform extract for such case.

```go
func extractOrEmbedImage(image *C.VipsImage, o Options) (*C.VipsImage, error) {
..
..
	case o.Top != 0 || o.Left != 0:
		if o.AreaWidth == 0 {
			o.AreaHeight = o.Width
		}
		if o.AreaHeight == 0 {
			o.AreaHeight = o.Height
		}
		if o.AreaWidth == 0 || o.AreaHeight == 0 {
			return nil, errors.New("Extract area width/height params are required")
		}
		image, err = vipsExtract(image, o.Left, o.Top, o.AreaWidth, o.AreaHeight)
		break
	}
..
..
}
```

What if we perform extract if:
- `o.AreaWidth != 0 || o.AreaHeight != 0`

or

- `o.Top != 0 || o.Left != 0 || o.AreaHeight != 0 || o.AreaWidth != 0`

```go
func extractOrEmbedImage(image *C.VipsImage, o Options) (*C.VipsImage, error) {
..
..
	case o.Top != 0 || o.Left != 0 || o.AreaHeight != 0 || o.AreaWidth != 0:
		if o.AreaWidth == 0 {
			o.AreaHeight = o.Width
		}
		if o.AreaHeight == 0 {
			o.AreaHeight = o.Height
		}
		if o.AreaWidth == 0 || o.AreaHeight == 0 {
			return nil, errors.New("Extract area width/height params are required")
		}
		image, err = vipsExtract(image, o.Left, o.Top, o.AreaWidth, o.AreaHeight)
		break
	}
..
..
}
```

**Comment by chonthu** [comment]
#70 Pull request was a test i created showing this doesn't work as expected

**Comment by h2non** [comment]
Gonna check that soon! Thanks

**Comment by h2non** [comment]
This should be fixed here: https://github.com/h2non/bimg/commit/3c158381b0d4ad06850cffc5be34984e3065f4c1

New version coming soon tomorrow!

**Comment by h2non** [comment]
Fixed in `v1.0.5`. Thanks for reporting.
